mongoDbTest.js

```diff
@@ -172,7 +183,48 @@ app.post('/delete-person', async (req, res) => {
 	}
 })
 
-
+app.post("/login", async (req, res) => {
+	const email = req.body.email;
+	const pwd = req.body.password;
+	let errMsg = null;
+	// Set response header regardless of success/failure
+	res.setHeader("Content-Type", "application/json");
+	try {
+		const results = await db.collection('users').find({emailAddress: email}).toArray();
+		if(results.length === 0) {
+			// Could not find user
+			res.status(401).send("Invalid email");
+		} else if (results.length > 1) {
+			// Many users with same email
+			// TODO: This is bad and should let someone know
+			errMsg = "Found multiple users with that email address.";
+			throw new Error();
+		} else {
+			// found user. validate password
+			bcrypt.compare(pwd, results[0].pwd, function(err, result) {
+				if(result) {
+					// Password matches, create session
+					req.session.loggedIn = true;
+					req.session.save(err => {
+						if(err)
+							console.log(err);
+					})
+					res.send("Logged in.");
+				} else {
+					// Password does not match
+					res.status(401).send("Invalid password");
+				}
+			});
+		}
+	} catch(e) {
+		console.log(e);
+		if(errMsg) {
+			res.status(403).send(errMsg);
+		} else {
+			res.status(500).send("There was most likely a problem with the server");
```

Comment:
What's the fail condition here, exactly? When would this code path be reached?

Reply:
![image](https://user-images.githubusercontent.com/37789629/166834202-b0157fdb-0be7-4d79-8730-568d2a6e5dc8.png)
![image](https://user-images.githubusercontent.com/37789629/166834266-16b9a1bf-5d5e-4d7e-80c4-2c88d39cc4ea.png)
![image](https://user-images.githubusercontent.com/37789629/166834487-8f460434-8c5a-4548-b26a-3e19f7d2d53f.png)


Currently the only case would be finding multiple user documents with the same email.

Reply:
I see! Worth documenting in the code, and sending back a more detailed error message if possible.